PxCs/Interface/PxSound.cs

```diff
@@ -0,0 +1,165 @@
+            }
+
+            var wavSound = PxVdf.pxVdfEntryOpenBuffer(vdfEntrySound);
+
+            if (wavSound == IntPtr.Zero)
+            {
+                throw new AccessViolationException("Sound could not be opened.");
+            }
+
+            try
+            {
+                ulong size = PxBuffer.pxBufferSize(wavSound);
+
+                var arrayPtr = IntPtr.Zero;
+                arrayPtr = PxBuffer.pxBufferArray(wavSound);
+                byte[] wavFile = new byte[size];
+
+                Marshal.Copy(arrayPtr, wavFile, 0, (int)size);
+                return wavFile;
+            }
+            finally
+            {
+                PxBuffer.pxBufferDestroy(wavSound);
+            }
+        }
+
+        public static float[] GetSoundFloatArrayFromVDF(IntPtr vdfPtr, string name)
+        {
+            byte[] wavFile = GetSoundByteArrayFromVDF(vdfPtr, name);
+            return ConvertWAVByteArrayToFloatArray(wavFile);
+        }
+
+        private static float[] ConvertWAVByteArrayToFloatArray(byte[] fileBytes)
+        {
+            byte[] fileBytesCopy = new byte[fileBytes.Length];
+            Array.Copy(fileBytes, fileBytesCopy, fileBytes.Length);
+
+            string riff = Encoding.ASCII.GetString(fileBytes, 0, 4);
+            string wave = Encoding.ASCII.GetString(fileBytes, 8, 4);
+            int subchunk1 = BitConverter.ToInt32(fileBytes, 16);
+            UInt16 audioFormat = BitConverter.ToUInt16(fileBytes, 20);
+
+            float[] data;
+
+            string formatCode = FormatCode(audioFormat);
+
+            UInt16 channels = BitConverter.ToUInt16(fileBytes, 22);
+            int sampleRate = BitConverter.ToInt32(fileBytes, 24);
+            int byteRate = BitConverter.ToInt32(fileBytes, 28);
+            UInt16 blockAlign = BitConverter.ToUInt16(fileBytes, 32);
+            UInt16 bitDepth = BitConverter.ToUInt16(fileBytes, 34);
+
+            int headerOffset = 16 + 4 + subchunk1 + 4;
+            int subchunk2 = BitConverter.ToInt32(fileBytes, headerOffset);
+
+            if (formatCode == "IMA ADPCM")
+            {
+                var ima = new Ima();
+                return ConvertWAVByteArrayToFloatArray(ima.Decode(fileBytesCopy));
+            }
+
+            switch (bitDepth)
+            {
+                case 8:
+                    data = Convert8BitByteArrayToFloatArray(fileBytes, headerOffset, subchunk2);
+                    break;
+                case 16:
+                    data = Convert16BitByteArrayToFloatArray(fileBytes, headerOffset, subchunk2);
+                    break;
+                default:
+                    throw new Exception(bitDepth + " bit depth is not supported.");
+            }
+
+            return data;
+        }
+
+        #region wav file bytes to Unity AudioClip conversion methods
+
+        private static float[] Convert8BitByteArrayToFloatArray(byte[] source, int headerOffset, int dataSize)
```

Comment:
How about having only one method which takes another parameter called from/to? With this you would decide from outside what byte[] type you send and want to retreive? Similar to this function: https://github.com/GothicKit/phoenix-csharp-interface/blob/main/PxCs/Interface/PxTexture.cs#L41